devito/ir/equations/equation.py

```diff
@@ -122,21 +122,34 @@ def __new__(cls, *args, **kwargs):
         oobs = detect_oobs(mapper)
         conditionals = [i for i in ordering if i.is_Conditional]
 
-        # The iteration space is constructed so that information always flows
-        # from an iteration to another (i.e., no anti-dependences are created)
-        iterators = build_iterators(mapper)
+        # Construct Intervals for IterationSpace and DataSpace
         intervals = build_intervals(Stencil.union(*mapper.values()))
-        intervals = IntervalGroup(intervals, relations=ordering.relations)
-        ispace = IterationSpace(intervals.zero(), iterators)
-
-        # The data space is relative to the computational domain. Note that we
-        # are deliberately dropping the intervals ordering (by turning `intervals`
-        # into a list), as this is irrelevant (even more: dangerous) for data spaces
-        intervals = [i if i.dim in oobs else i.zero() for i in intervals]
-        intervals += [Interval(i, 0, 0) for i in ordering
-                      if i not in ispace.dimensions + conditionals]
-        parts = {k: IntervalGroup(build_intervals(v)) for k, v in mapper.items() if k}
-        dspace = DataSpace(intervals, parts)
+        iintervals = []
```

Comment:
I am not sure if I like iintervals varname... just saying...

Reply:
I can't afford longer variable names otherwise I have to break to many lines and it gets horrible

But you made a fair point: I added comments clarifying what they represent